### PR TITLE
🤠 Avatar

### PR DESCRIPTION
**Pull request by adnan1naeem**
# 🤠 Avatar

Avatar control enables users to add images, icons or initials
within any application for instances such as a profile pictures.

## Usage

```jsx
<Avatar type="image" image={{ uri: 'https://picsum.photos/200' }} />
```

## Screenshots

Add screenshots here

## Supported Props

> Legend
    ✅ = Fully Compatible and works as intended
    ⚠️ = Partially compatible
    ❌ = Not compatible

|  Prop | Compatible |           Comments           |
| :---: | :--------: | :--------------------------: |
|  type |     ️⚠️    | Only works with `image` type |
|  size |     ❌ ️    |                              |
| color |     ❌ ️    |                              |
|  icon |     ❌ ️    |                              |
| image |     ❌ ️    |                              |
|  text |     ❌ ️    |                              |
| style |     ❌ ️    |                              |

## User Stories

|                              Story                              | In Storybook | Has Unit Test |
| :-------------------------------------------------------------: | :----------: | :-----------: |
|  User should be able to use Image Avatar by setting image path  |       ✅      |       ✅       |
| User should be able to use Image Avatar importing image in code |       ❌      |       ❌       |

### CODE REVIEW

**Comment by adnan1naeem** [comment]
all stories working but dont know why build is failing ?

File: package.json [hunk 1]
```diff
@@ -122,6 +122,7 @@
     "enzyme-async-helpers": "^0.9.1",
     "gh-pages": "^2.0.1",
     "jest": "^24.3.1",
+    "react-native": "0.55.4",
```
Comment (artalat) [inline package.json]: why did you add react native?

File: src/components/Avatar/Avatar.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import MuiAvatar from '@material-ui/core/Avatar';
+import React from 'react';
+
+export const Avatar = (props: any) => {
+
+	const source = props.image && props.image.uri ? props.image.uri : props.image;
```
Comment (artalat) [inline src/components/Avatar/Avatar.tsx]: This will not work in all situations. For example if someone does this:

```jsx
import img from './path/to/image.jpg`

<Avatar type="image" image={img} />
```

This WILL fail!

File: src/components/Avatar/Avatar.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+	return (
+		<MuiAvatar  src={source} {...props} />
```
Comment (artalat) [inline src/components/Avatar/Avatar.tsx]: What about other avatar types? icon, text, etc?

File: src/components/Avatar/_stories_/Avatar.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+
+const Avatar = getComponent<AvatarProps>('Avatar');
+
+const stories = storiesOf('Avatar', module);
+
+stories.addDecorator(withInfo);
+stories.addDecorator(withKnobs);
+
+stories.add('Simple Avatar ', () => (
```
Comment (artalat) [inline src/components/Avatar/_stories_/Avatar.stories.tsx]: Add stories and tests for other avatar types as well. i.e. icon and text.

File: src/components/Avatar/Avatar.tsx (hunk 2 above)
Comment (adnan1naeem) [reply src/components/Avatar/Avatar.tsx]: what should be the option for it kindly provide some code

File: src/components/Avatar/_stories_/Avatar.stories.tsx [hunk 5]
```diff
@@ -19,11 +19,26 @@ stories.add('Simple Avatar ', () => (
...
 		image={{ uri: 'https://s3-us-west-2.amazonaws.com/bluerainimages/water-dispenser.svg' }}
 	/>
+));
+
+
 
+stories.add('Simple Avatar with text prop ', () => (
+	<Avatar
+		type="text"
+		text={{ uri: 'hello' }}   // why this prop is of imagesource type ? issue in typings.
```
Comment (adnan1naeem) [inline src/components/Avatar/_stories_/Avatar.stories.tsx]: text prop should be of String type according to me

**Comment by adnan1naeem** [comment]
code is in this pull request #42
